src/Init/Data/Int/DivModLemmas.lean

```diff
@@ -1348,3 +1358,7 @@ theorem bmod_natAbs_plus_one (x : Int) (w : 1 < x.natAbs) : bmod x (x.natAbs + 1
           all_goals decide
     · exact ofNat_nonneg x
     · exact succ_ofNat_pos (x + 1)
+
+theorem bmod_neg_bmod : bmod (-(bmod x n)) n = bmod (-x) n := by
```

Comment:
Also just a style question more than anything, since this simplifies the statement, should this be a `@[simp]` lemma, even though it's probably a rare case? It's not entirely clear to me when to apply @[simp], since some lemmas that are simplifying are not marked in lean4/mathlib and I haven't figured out the exact rule as to when (outside of obvious considerations about normal forms.)

Reply:
Yes, let's add `@[simp]` here. 

Ideally all lemmas that look like good candidates for `simp`, but for some reason are not, should have a comment (usually a plain comment, not a doc-comment) explaining this. But we are a long way from that ideal.